source/extensions/common/wasm/context.cc

```diff
@@ -902,6 +905,15 @@ WasmResult Context::httpCall(absl::string_view cluster, const Pairs& request_hea
   // Check that we were provided certain headers.
   if (message->headers().Path() == nullptr || message->headers().Method() == nullptr ||
       message->headers().Host() == nullptr) {
+    if (message->headers().Path() == nullptr) {
+      ENVOY_LOG(trace, "wasm log{}: path required", log_prefix());
+    }
+    if (message->headers().Method() == nullptr) {
+      ENVOY_LOG(trace, "wasm log{}: method required", log_prefix());
+    }
+    if (message->headers().Host() == nullptr) {
+      ENVOY_LOG(trace, "wasm log{}: host required", log_prefix());
+    }
```

Comment:
consolidate this to a one line log to something like:

ENVOY_LOG(trace, "wasm log{}: {}{}{}required", log_prefix(), message->headers().Path() == nullptr ? "path" : "", ...);